aiopg/sa/result.py

```diff
@@ -121,9 +120,7 @@ def __init__(self, result_proxy, metadata):
             name, obj, type_ = (
                 colname,
                 None,
-                result_map.get(
-                    colname,
-                    typemap.get(coltype, sqltypes.NULLTYPE))
```

Comment:
We can do `hasattr(dialect, 'dbapi_type_map')` check, and make code working with both version of sqlalchemy

Reply:
if we want to keep `dbapi_type_map` compatibility, I suggest something like:
```
typemap = getattr(dialect, 'dbapi_type_map', {})
```
and keep `result_map` as is.

Reply:
@jettify if you prefer this way I can fix my changes

Reply:
Looks good, please also add comment explaining why we do that hack

Reply:
Done, please review